zipline/utils/numpy_utils.py

```diff
@@ -493,3 +494,11 @@ def changed_locations(a, include_first):
         return indices
 
     return hstack([[0], indices])
+
+
+def compare_datetime_arrays(x, y):
+    """
+    Compare datetime64 ndarrays in a NaT-aware fashion.
+    """
+
+    return array_equal(x.view('int64'), y.view('int64'))
```

Comment:
Is there a reason to use `array_equal` over `==` here?

Reply:
`array_equal` also checks for matching shapes. If we use `==` we could run into an issue where arrays of different length containing the same value compare as equal:
```
In [18]: a = np.array(['2014-01-01', '2014-01-01'], dtype='M8[ns]')

In [19]: b = np.array(['2014-01-01'], dtype='M8[ns]')

In [20]: a.view('int64') == b.view('int64')
Out[20]: array([ True,  True], dtype=bool)
```